casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -826,12 +838,35 @@ object BlockAPI {
                            .pure[F]
                        case None =>
                          for {
-                           lastFinalizedBlock <- casper.lastFinalizedBlock
-                           runtimeManager     <- casper.getRuntimeManager
-                           postStateHash      = ProtoUtil.postStateHash(lastFinalizedBlock)
-                           res                <- runtimeManager.playExploratoryDeploy(term, postStateHash)
-                           lightBlockInfo     <- getLightBlockInfo[F](lastFinalizedBlock)
-                         } yield (res, lightBlockInfo).asRight[Error]
+                           targetBlock <- if (blockHash.isEmpty)
+                                           casper.lastFinalizedBlock.map(_.some)
+                                         else
+                                           for {
+                                             hashByteString <- Base16
+                                                                .decode(blockHash.getOrElse(""))
+                                                                .map(ByteString.copyFrom)
+                                                                .liftTo[F](
+                                                                  BlockRetrievalError(
+                                                                    s"Input hash value is not valid hex string: $blockHash"
+                                                                  )
```

Comment:
It's ok to throw this error here because it's handled here and converted to ApiErr.
https://github.com/rchain/rchain/blob/bf4cc6bf94542bfc919e0a0db6c00b28621a7a4e/casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala#L596-L604